src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts

```diff
@@ -31,17 +39,35 @@ export class CreateCustomerComponent implements OnInit, OnDestroy {
     this.saveSubscription = messages$.subscribe((valueMessage) => {
       this.setStatusOnScreen(valueMessage);
     });
+
+    const customers$ = this.store.pipe(select(getCustomerById));
+    this.editSubscription = customers$.subscribe((customer) => {
+      this.customerToEdit = customer;
+      this.setDataToUpdate(this.customerToEdit);
+    });
   }
 
   ngOnDestroy() {
     this.areTabsActive = false;
     this.saveSubscription.unsubscribe();
+    this.editSubscription.unsubscribe();
   }
 
   onSubmit(customerData) {
-    this.store.dispatch(new CreateCustomer(customerData));
-    this.store.dispatch(new LoadCustomers());
+    const key = 'id';
+    if (this.customerToEdit) {
+      const customer = {
+        ...customerData,
+        id: this.customerToEdit[key],
```

Comment:
I think a better solution would be this.
`const customer = { ...this.customerToEdit, ...customerData };` o  `const customer = { ...customerData, id: this.customerToEdit.id };`
with this we would eliminate line 57
`const key = 'id';`

Reply:
done